pkg/networkmanager/interfaces.js

```diff
@@ -785,18 +824,25 @@
             connections_by_uuid[settings.connection.uuid] = obj;
     }
 
-    function refresh_settings(obj) {
+    async function refresh_settings(obj) {
         push_refresh();
-        client.call(objpath(obj), "org.freedesktop.NetworkManager.Settings.Connection", "GetSettings")
-                .then(function(reply) {
-                    const result = reply[0];
-                    if (result) {
-                        priv(obj).orig = result;
-                        set_settings(obj, settings_from_nm(result));
-                    }
-                })
-                .catch(complain)
-                .finally(pop_refresh);
+
+        try {
+            const reply = await client.call(objpath(obj), "org.freedesktop.NetworkManager.Settings.Connection", "GetSettings");
+            const result = reply[0];
+            if (result) {
+                if (result.wireguard) {
+                    const secrets = await client.call(objpath(obj), "org.freedesktop.NetworkManager.Settings.Connection", "GetSecrets", ["wireguard"]);
+                    result.wireguard['private-key'] = secrets[0].wireguard['private-key'];
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230703-110332-55dbfd0f-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)